src/main/java/core/basesyntax/Main.java

```
@@ -0,0 +1,65 @@
+import core.basesyntax.model.operation.PurchaseOperation;
+import core.basesyntax.model.operation.ReturnOperation;
+import core.basesyntax.model.operation.SupplyOperation;
+import core.basesyntax.service.DataConverter;
+import core.basesyntax.service.FileReader;
+import core.basesyntax.service.FileWriter;
+import core.basesyntax.service.ReportGenerator;
+import core.basesyntax.service.ShopService;
+import core.basesyntax.service.impl.CsvDataConverterImpl;
+import core.basesyntax.service.impl.CsvFileReaderImpl;
+import core.basesyntax.service.impl.CsvFileWriterImpl;
+import core.basesyntax.service.impl.ReportGeneratorImpl;
+import core.basesyntax.service.impl.ShopServiceImpl;
+import core.basesyntax.strategy.OperationStrategy;
+import core.basesyntax.strategy.OperationStrategyImpl;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class Main {
+    private static final String FILES_PATH = "src/main/resources/";
+    private static final String INPUT_FILE_NAME = "input.csv";
+    private static final String OUTPUT_FILE_NAME = "output.csv";
+    private static final String INPUT_FILE_PATH = FILES_PATH + INPUT_FILE_NAME;
+    private static final String OUTPUT_FILE_PATH = FILES_PATH + OUTPUT_FILE_NAME;
+
+    public static void main(String[] args) {
+        // 1. Read the data from the input CSV file
+        FileReader fileReader = new CsvFileReaderImpl();
+        List<String> inputReport = fileReader.read(INPUT_FILE_PATH);
+
+        // 3. Create a storage for managing how many products quantity available
+        Storage storage = Storage.getInstance();
+
+        // 4. Create and fill the map with all OperationHandler implementations
+        Map<FruitTransaction.Operation, OperationHandler> operationHandlers = new HashMap<>();
+        operationHandlers.put(FruitTransaction.Operation.BALANCE, new BalanceOperation(storage));
+        operationHandlers.put(FruitTransaction.Operation.PURCHASE, new PurchaseOperation(storage));
```

The `PurchaseOperation` class should check the resulting balance before updating the `Storage`. If the balance becomes negative, it should throw a `RuntimeException` to prevent invalid states.